libbluray: enable all features by default

shelvacu opened the pull request:
withJava was originally defaulted to false in 1a4bb0ef9b4abc14325541ce2f145c364838de12 the closure size is no longer significantly larger (41MB vs 45MB)

withAACS has defaulted to false since the package's inception in 2012 with no explanation. 383ec69fb68ed9782f2dc758bd479680f233a341

withBDplus has defaulted to false since it was added in 2015 with no explanation. a5fc6f702e0ed5fca2b171ea3c5dcb6f9da8dfb4


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

shelvacu committed:
libbluray: enable all features by default

withJava was originally defaulted to false in 1a4bb0ef9b4abc14325541ce2f145c364838de12
the closure size is no longer significantly larger (41MB vs 45MB)

withAACS has defaulted to false since the package's inception in 2012
with no explanation. 383ec69fb68ed9782f2dc758bd479680f233a341

withBDplus has defaulted to false since it was added in 2015
with no explanation. a5fc6f702e0ed5fca2b171ea3c5dcb6f9da8dfb4

clhodapp commented:
Note: There's now `libbluray-full`, which enables these flags.

My intention had been to start going through various media programs and switching them to the `full` version as appropriate.

shelvacu committed:
libbluray-full: alias to libbluray

libluray now has all features enabled by default (see parent commit), so libbluray-full is no longer needed.
